apps/cowswap-frontend/src/modules/swap/hooks/useIsEthFlowOrderExists.ts

```diff
@@ -0,0 +1,39 @@
+import { useAtomValue } from 'jotai'
+import { useCallback } from 'react'
+
+import { ZERO_ADDRESS } from 'legacy/constants/misc'
+import { useEthFlowContract } from 'legacy/hooks/useContract'
+
+import { ethFlowInFlightOrderIdsAtom } from '../state/EthFlow/ethFlowInFlightOrderIdsAtom'
+
+export interface EthFlowOrderExistsCallback {
+  (orderId: string, orderDigest: string): Promise<boolean>
+}
+
+export function useIsEthFlowOrderExists(): EthFlowOrderExistsCallback {
+  const ethFlowInFlightOrderIds = useAtomValue(ethFlowInFlightOrderIdsAtom)
+  const ethFlowContract = useEthFlowContract()
+
+  return useCallback(
+    async (orderId: string, orderDigest: string) => {
+      if (ethFlowInFlightOrderIds.includes(orderId)) {
+        return true
+      }
+
+      if (ethFlowContract) {
+        try {
+          const { owner } = await ethFlowContract.callStatic.orders(orderDigest)
+
+          return owner.toLowerCase() !== ZERO_ADDRESS
```

Comment:
Don't we need to check the owner matches the one placing the order?

Reply:
Yes and no at the same time :)
If the smart-contract storage contains an order with the same id, it means there is a collision.
How orders map looks like:
```
mapping(bytes32 => EthFlowOrder.OnchainData) public orders;
```

It's actually key-value, where key is `orderDigest`. And here I see smth interesting, `orderDigest` is a hash of an order that doesn't contain the owner address implicitly. It contains it as a `receiver` but the receiver can be any address.
So, theoretically, two different accounts can create an order with the same `orderDigest` and in this key the last one will face a collision.
In conclusion, I think we should take into account even this case and check if `owner` is not zero.

<img width="529" alt="image" src="https://github.com/cowprotocol/cowswap/assets/7122625/6afcd5c8-f837-4246-8fb2-dfadf005b715">

Reply:
```
if (orders[orderHash].owner != EthFlowOrder.NO_OWNER) {
            revert OrderIsAlreadyOwned(orderHash);
        }
```

Even in the smart-contact they do the same check